### PR TITLE
Pcm 992

### PR DESCRIPTION
**Pull request by renujhamtani**
@sshumake @pbathia This is first pass for case discussion using uds services. This is not yet fully complete. But pranjal needs some basic parts for her issue. Please merge this whenever reviewed and done.

### CODE REVIEW

File: app/ascension/controllers/addComments.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        $scope.addingattachment = false;
+        $scope.progressCount = 0;
+        $scope.charactersLeft = 0;
+        $scope.maxCommentLength = '32000';
+        $scope.ieFileDescription = '';
+
+        CaseDiscussionService.commentTextBoxEnlargen = false;
+
+        $scope.clearComment = function(){
+            CaseDetailsService.commentText = '';
```
Comment (pbathia) [inline app/ascension/controllers/addComments.js]: here, in my opinion assigning as 'CaseDetailService.commentText = undefined' would be good practice.

File: app/ascension/controllers/addComments.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                if(securityService.loginStatus.authedUser.sso_username !== undefined && CaseDetailsService.updatedNotifiedUsers.indexOf(securityService.loginStatus.authedUser.sso_username) === -1){
+                    strataService.cases.notified_users.add(caseNumber, securityService.loginStatus.authedUser.sso_username).then(function () {
+                        CaseDetailsService.updatedNotifiedUsers.push(securityService.loginStatus.authedUser.sso_username);
+                    }, function (error) {
+                        AlertService.addStrataErrorMessage(error);
+                    });
+                }
+
+            };
+            var onError = function (error) {
```
Comment (pbathia) [inline app/ascension/controllers/addComments.js]: I think you have not used this, if not planning to use in next pass, should be removed.

File: app/ascension/controllers/addComments.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                $scope.addingComment = false;
+                $scope.progressCount = 0;
+                $scope.charactersLeft = 0;
+            };
+
+            $scope.addingComment = true;
+
+            if(CaseDetailsService.isCommentPublic)
+            {
+               udsService.kase.comments.post.public(CaseDetailsService.kase.case_number,CaseDetailsService.commentText);
```
Comment (pbathia) [inline app/ascension/controllers/addComments.js]: I think here error condition should be handled using promise.

File: app/ascension/controllers/addComments.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            else if(CaseDetailsService.commentText===undefined)
+            {
+                $scope.progressCount=0;
+                $scope.charactersLeft = 0;
+            }
+        };
+        $scope.saveDraft = function () {
+        };
+        $scope.shouldTextboxMinimize = function(){
+            if(CaseDetailsService.commentText === undefined || CaseDetailsService.commentText === ''){
```
Comment (pbathia) [inline app/ascension/controllers/addComments.js]: here, you can make use of RHAUtils.isEmpty function

File: app/ascension/views/addCommentSection.jade [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        .pull-right
+            .spinner.spinner-inline(ng-show='addingComment || addingattachment')
+            button#rha-case-addcommentbutton.btn.btn-app(
+            ng-hide='addingComment || addingattachment',
+            ng-disabled='!ie8 && !ie9 && (CaseDetailsService.disableAddComment && AttachmentsService.updatedAttachments.length === 0 && !AttachmentsService.hasBackEndSelections())',
+            ng-click='addComment()',
+            translate='') Post
+            a.margin-left#rha-case-addcommentbutton(ng-click='clearComment()',translate='',ng-hide='addingComment || addingattachment') Cancel
+.row
+    .col-xs-12(ng-show='ie8 && EDIT_CASE_CONFIG.isPCM || ie9 && EDIT_CASE_CONFIG.isPCM')
```
Comment (pbathia) [inline app/ascension/views/addCommentSection.jade]: I dont think we need this as we are not planning to use IE8/9 for ascension. We should remove this code to make it simpler.  Still we can have discussion with Spenser first.

File: app/ascension/controllers/attachLocalFile.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            });
+            $scope.clearSelectedFile();
+            if ($scope.$root.$$phase !== '$apply' && $scope.$root.$$phase !== '$digest') {
+                $scope.$apply();
+            }
+        };
+        $scope.getFile = function () {
+            $('#fileUploader').click();
+        };
+        $scope.selectFile = function () {
```
Comment (pbathia) [inline app/ascension/controllers/attachLocalFile.js]: i think this function we have used for IE support, we should clean IE related code to make it simple.

File: app/ascension/controllers/caseDetails.js [hunk 7]
```diff
@@ -7,10 +7,15 @@ angular.module('RedhatAccess.ascension').controller('CaseDetails', [
     'AUTH_EVENTS',
     'TOPCASES_EVENTS',
     'RHAUtils',
-    function ($scope, CaseDetailsService, securityService, AUTH_EVENTS, TOPCASES_EVENTS, RHAUtils) {
+    '$sce',
```
Comment (pbathia) [inline app/ascension/controllers/caseDetails.js]: are you planning to use this in next pass?

File: app/common/services/udsService.js [hunk 8]
```diff
@@ -39,6 +39,19 @@ angular.module('RedhatAccess.common').factory('udsService', [
                 kase.account.account_number = response.resource.account.resource.accountNumber;
                 kase.account.is_strategic = response.resource.account.resource.strategic;
                 kase.account.special_handling_required = response.resource.account.resource.specialHandlingRequired;
+                kase.attachments={};
+                kase.attachments=response.resource.fileAttachments;
+
+                angular.forEach( kase.attachments, angular.bind(this, function (attachment) {
+                    var lastModifiedDate = RHAUtils.convertToTimezone(attachment.resource.lastModified);
+                    var modifiedDate = attachment.resource.lastModified;
+                    attachment.resource.sortModifiedDate = modifiedDate;
```
Comment (pbathia) [inline app/common/services/udsService.js]: you can directly assign to attachment.resource.lastModified, as you are not using this modifiedDate further and not doing any formatting/conversion also.

**Comment by pbathia** [comment]
I am not very sure about backend attachment ? GSS guys attach such files ? We need to get business scenario for this and then take appropriate action for backend attachment code.

File: app/ascension/controllers/addComments.js (hunk 2 above)
Comment (renujhamtani) [inline app/ascension/controllers/addComments.js]: I have used this but not sure whether to use strata for notifications. It is being called for adding coments. Hence i have pout a note.

**Comment by renujhamtani** [comment]
@pbathia  : The whole attachment code is not working. I pushed all the files as you wanted it to be done. I have written that only the comments section works (that also without draft comments). So please ignore these for attachments..

File: app/ascension/controllers/addComments.js (hunk 3 above)
Comment (renujhamtani) [inline app/ascension/controllers/addComments.js]: Pranjal, not sure on this. Same way hs been used in PCM code.

File: app/ascension/views/addCommentSection.jade (hunk 5 above)
Comment (renujhamtani) [inline app/ascension/views/addCommentSection.jade]: I will remove this once i start working on atttachment stuff,,,,

File: app/ascension/controllers/attachLocalFile.js (hunk 6 above)
Comment (renujhamtani) [inline app/ascension/controllers/attachLocalFile.js]: I will remove this once i start working on atttachment stuff,,,,

File: app/ascension/controllers/caseDetails.js (hunk 7 above)
Comment (renujhamtani) [inline app/ascension/controllers/caseDetails.js]: removed the unnnecesary lines..

File: app/common/services/udsService.js (hunk 8 above)
Comment (renujhamtani) [inline app/common/services/udsService.js]: done

**Comment by renujhamtani** [comment]
@pbathia I have incorporated your comments for "comments" section. Attachments (i am yet to work). The error message i will look with the attachments one. I think you can start working on ur issues as the basic layout is complete. I will merge this once you give a go ahead.

**Comment by pbathia** [comment]
We can merge.

**Pull request merged**